test/tests/parallel/set3/mta/dsn.bats

```diff
@@ -0,0 +1,88 @@
+load "${REPOSITORY_ROOT}/test/helper/setup"
+load "${REPOSITORY_ROOT}/test/helper/common"
+
+BATS_TEST_NAME_PREFIX='[DSN] '
+CONTAINER_NAME='dms-test_dsn'
+
+function teardown() { _default_teardown ; }
+
+@test "should always send a DSN when requested" {
+
+  local LOG_DSN='delivery status notification'
+  local CONTAINER_ARGS_ENV_CUSTOM=(
+    # Required only for delivery via nc (_send_email)
+    --env PERMIT_DOCKER=container
+  )
+
+  _init_with_defaults
+  # Unset `smtpd_discard_ehlo_keywords` to allow DSNs by default on any `smtpd` service:
+  mv "${TEST_TMP_CONFIG}/dsn/postfix-main.cf" "${TEST_TMP_CONFIG}/postfix-main.cf"
+  _common_container_setup 'CONTAINER_ARGS_ENV_CUSTOM'
+
+  _send_email 'email-templates/dsn-unauthenticated'
+  _send_email 'email-templates/dsn-authenticated' '0.0.0.0 465'
+  _send_email 'email-templates/dsn-authenticated' '0.0.0.0 587'
+  _wait_for_empty_mail_queue_in_container
```

Comment:
Very good, but checking here is the best way of doing it:

```suggestion
export MAIL_ID1=$(_send_email_and_get_id 'email-templates/dsn-unauthenticated')
  export MAIL_ID2=$(_send_email_and_get_id email-templates/dsn-authenticated' '0.0.0.0 465')
  export MAIL_ID3=$(_send_email_and_get_id 'email-templates/dsn-authenticated' '0.0.0.0 587')

  for ID in MAIL_ID{1,2,3}; do
    [[ -n ${!ID} ]] || { echo "${ID} is empty - aborting!" ; return 1 ; }
  done
```

This way, you can later use

```bash
_print_mail_log_for_id "${MAIL_ID1}"
```

and run asserts on the log for this email.